### DISCUSSION TITLE
Undiscoverable app for devtools

### DISCUSSION THREAD

**Discussion by davidbrochart** [Textualize/textual#1186] [Q&A] [answered]
In [jpterm](https://github.com/davidbrochart/jpterm), the Textual [app](https://github.com/davidbrochart/jpterm/blob/c0be2bbbae4b910a643f896ad1e82cef5ac99ea5/plugins/jpterm/txl_jpterm/components.py) is a plugin which is eventually required by [Asphalt](https://asphalt.readthedocs.io) and then [run](https://github.com/davidbrochart/jpterm/blob/937142055bd32d622c17ae24cf15ef947ec5b4c6/txl/txl/app.py#L28-L29). It doesn't seem to be possible to use the devtools with something like:
```console
textual run my_app.py:alternative_app
```
I was wondering if there was a lower-level API to the devtools that would work in this case?

**Comment by davep** [comment, marked as answer]
Apologies if I've not fully understood the way your app works, but another way of getting a Textual app to talk to the `devtools` is to set `TEXTUAL=devtools` in your environment -- I use that in some situations where it wouldn't be convenient to use `textual run`. Might that work for you?

PS: Don't forget that when using `textual run` to talk to devtools you'd include `--dev`.

**Reply by davidbrochart** [reply]
Awesome, thanks @davep!

**Reply by davep** [reply]
My pleasure. Glad to help. :-)

**Reply by davidbrochart** [reply]
It seems that this way of getting a Textual app to talk to the `devtools` is not listed in the [documentation](https://textual.textualize.io/guide/devtools/)? Each time I have to look it up I need to find this discussion again :smile:
